### DISCUSSION TITLE
Error: 'No labels found in /home/zl481/PycharmProjects/pythonProject1/datasets/VisDrone/VisDrone2019-DET-train/labels.cache.'

### DISCUSSION THREAD

**Discussion by 1nyourlife** [ultralytics/ultralytics#2808] [Q&A]
I download all the files I need, and write this function:
![image](https://github.com/ultralytics/ultralytics/assets/95243930/1117e678-68dd-47fc-b26b-e1185088b371)
![image](https://github.com/ultralytics/ultralytics/assets/95243930/7275e9c0-3ac9-4c55-a0da-8d6ff7313ffc)
Get an error:
![image](https://github.com/ultralytics/ultralytics/assets/95243930/39ca8777-9869-48b1-b92c-4a8e4ba786c7)

If I use coco128.yaml, it works, but when I use VisDrone.yaml, it failed.
I do not change anything in VisDrone.yaml, and download the dataset from the given links. I can not find out why it comes an error.

**Comment by glenn-jocher** [comment]
@1nyourlife hello! It seems like you're encountering an issue where the labels for your VisDrone dataset are not being found. This is typically due to one of the following reasons:

1. The path to the labels in your dataset YAML file might be incorrect. Please double-check that the paths specified for the train, val, and test sets are correct and that they point to the directories where your label files are located.

2. The label files might be missing or in an incorrect format. Ensure that the label files are present in the specified directories and that they are in the correct format expected by YOLOv8.

3. The `.cache` file might be outdated or corrupted. Try deleting the `labels.cache` file, and YOLOv8 will automatically recreate it when you run the training again.

Make sure that your dataset is structured properly, with images and labels in the corresponding folders as outlined in the documentation. If you continue to face issues, you might want to review the dataset setup section in our documentation for more detailed instructions on how to properly configure your custom dataset.

If the problem persists after checking these points, please provide more details about your dataset structure and the contents of your VisDrone.yaml file for further assistance. 🛠️
